lib/vnet/diag/diagnostician.go

```diff
@@ -0,0 +1,135 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package diag
+
+import (
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	diagv1 "github.com/gravitational/teleport/gen/proto/go/teleport/lib/vnet/diag/v1"
+)
+
+// Diagnostician runs individual diag checks along with their accompanying commands and produces a
+// report.
+type Diagnostician struct{}
```

Comment:
It does not look like this type exists to satisfy any particular interface, so I'm not sure why this type exists. The methods could just be plain functions? Or do you expect this struct to grow some fields in the near future?

Reply:
I used it to group similar functions, those that execute checks rather than doing any checking themselves. I've thought that it's a common practice in Go. But now that I think about it, probably most of the times I saw an empty struct it was used for some specific interface. And many packages I saw in the wild do have a few public functions that work on the structs exported by the package.

I'll rewrite this to regular functions.